modules/tinymce/src/themes/silver/main/ts/ui/menus/menu/SingleMenu.ts

```diff
@@ -97,32 +98,60 @@ export const createAutocompleteItems = (
   onItemValueHandler: (itemValue: string, itemMeta: Record<string, any>) => void,
   columns: 'auto' | number,
   itemResponse: ItemResponse,
-  sharedBackstage: UiFactoryBackstageShared
+  sharedBackstage: UiFactoryBackstageShared,
+  highlight: string[]
 ) => {
   // Render text and icons if we're using a single column, otherwise only render icons
   const renderText = columns === 1;
   const renderIcons = !renderText || MenuUtils.menuHasIcons(items);
   return Optionals.cat(
     Arr.map(items, (item) => {
-      if (item.type === 'separator') {
-        return InlineContent.createSeparatorItem(item).fold(
-          MenuUtils.handleError,
-          (d) => Optional.some(MenuItems.separator(d))
-        );
-      } else {
-        return InlineContent.createAutocompleterItem(item).fold(
-          MenuUtils.handleError,
-          (d: InlineContent.AutocompleterItem) => Optional.some(MenuItems.autocomplete(
-            d,
-            matchText,
-            renderText,
-            'normal',
-            onItemValueHandler,
-            itemResponse,
-            sharedBackstage,
-            renderIcons
-          ))
-        );
+      switch (item.type) {
+        case 'separator':
+          return InlineContent.createSeparatorItem(item).fold(
+            MenuUtils.handleError,
+            (d) => Optional.some(MenuItems.separator(d))
+          );
+
+        case 'cardmenuitem':
+          return BridgeMenu.createCardMenuItem(item).fold(
+            MenuUtils.handleError,
+            (d) => Optional.some(MenuItems.card(
+              {
+                ...d,
+                // Intercept action
+                onAction: (api) => {
+                  onItemValueHandler(d.value, d.meta);
```

Comment:
When we discussed this last week, we said this should be the opposite order. As it'd be odd to have the generic handler fire before the handler that's specific to the. menu item.